tests/transform_dialect/cuda/mma_using_layout_analysis.mlir

```diff
@@ -0,0 +1,20 @@
+// RUN: iree-opt %s --split-input-file --iree-transform-dialect-interpreter | FileCheck %s
+
+func.func @matmul(%lhs : tensor<16x16xf16>, %rhs : tensor<16x8xf16>) -> tensor<16x8xf16> {
+  %c0 = arith.constant 0.0 : f16
+  %0 = tensor.empty() : tensor<16x8xf16>
+  %1 = linalg.fill ins(%c0 : f16) outs(%0 : tensor<16x8xf16>) -> tensor<16x8xf16>
+  %2 = linalg.matmul ins(%lhs, %rhs : tensor<16x16xf16>, tensor<16x8xf16>)
+      outs(%1 : tensor<16x8xf16>) -> tensor<16x8xf16>
+  return %2 : tensor<16x8xf16>
+}
+
+// RUN: iree-compile %s --iree-hal-target-backends=cuda \
+// RUN:     --iree-codegen-llvmgpu-enable-transform-dialect-jit=false \
+// RUN:     --iree-codegen-llvmgpu-use-transform-dialect=%p/reduction_v2_codegen_spec.mlir | \
+// RUN: iree-run-module --function=matmul --device=cuda --input="16x16xf16=1" --input="16x8xf16=2" |\
```

Comment:
Should we have more complex inputs to make sure we are reading the right elements? We won't be able to tell if we are loading things right with this case?

Reply:
Sure I can increase the tensor sizes.

Reply:
I meant keep the size but have non uniform input?

Reply:
Oh I see, yes sure. I can add non-uniform input.

Reply:
So what's the best way to add non-uniform input? I think numpy files would be ideal here but not sure how to integrate them with the BUILD script. Or do you have any other ideas?

Reply:
here is an example:
https://github.com/openxla/iree/blob/817c771c783669d7a5cfaa2b1199f3ed58b1c305/tests/e2e/regression/transpose.mlir#L18

Reply:
Thanks! I have updated the patch with a non uniform test. Please merge when you can as I don't have merge privileges.